tools/check-nuttx-release.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+URL=$1
+
+function download_release() {
```

Comment:
You can probably use the wget recursive function to do this for you for example
```
wget -r -np -nH --cut-dirs=5 -R "index.html*" https://dist.apache.org/repos/dist/dev/incubator/nuttx/9.1.0-RC1/
❯ tree
.
├── 9.1.0-RC1
│   ├── apache-nuttx-9.1.0-incubating.tar.gz
│   ├── apache-nuttx-9.1.0-incubating.tar.gz.asc
│   ├── apache-nuttx-9.1.0-incubating.tar.gz.sha512
│   ├── apache-nuttx-apps-9.1.0-incubating.tar.gz
│   ├── apache-nuttx-apps-9.1.0-incubating.tar.gz.asc
│   └── apache-nuttx-apps-9.1.0-incubating.tar.gz.sha512
└── robots.txt
```

Reply:
@btashton Good idea, thanks. I updated the script. I had to also use `--cut-dirs` to avoid a deeply nested directory hierarchy.

Reply:
@btashton If you're happy with this, I'll rebase and force push, let me know. Or else let me know if you want to see other improvements.

Reply:
@adamfeuer Yeah sorry I forgot that and updated it later.  There is also the `-nH` flag I added, but it looks like you are using the host as the temporary directory, so that works as well.  Any reason to keep the full URL as the input instead of just the release name?  Anyway I'll mark this as approved and leave it to your discretion.  We should probably also consider adding some traps or a lazy `set -e` to capture the failure.

Reply:
@btashton I just used the URL since that's what you posted in your email :slightly_smiling_face:  I'm open to change it, let me know if you want that.

Re: traps, yeah, I think so to. This just started as a quick and dirty way to avoid manual work... so how about we do the traps in another PR?

Re: lazy `set -e`, what do you mean? I think I don't understand.

Reply:
if you use `set -e` then if any of the commands have a non 0 status code the script will fail.  Not all commands return 0 on success, so it is a way to lazy handle it if they do.  The URL is fine, we can make this better in a future release, so lets start here :)

Reply:
@btashton Ah, I just didn't understand what you meant by "lazy". I'll add `set -e` and force push. Yes, let's get it in and make it better later.

Reply:
I like the idea of passing only the release name, the rest of the URL doesn't change.

Reply:
@Ouss4 Ok, I added the ability to do either URL or release name.
@btashton I added the trap statement too.

Ok, will force push.